packages/ui/src/components/va-split/VaSplit.vue

```diff
@@ -0,0 +1,261 @@
+<template>
+  <section
+    ref="splitPanelsContainer"
+    aria-label="split panels"
+    class="va-split"
+    :class="classComputed">
+    <div
+      class="va-split__panel"
+      :style="getPanelStyle('start')">
+      <slot name="start" />
+    </div>
+    <div
+      class="va-split__dragger"
+      :style="draggerStyleComputed"
+      @mousedown="startDragging"
+      @touchstart="startDragging"
+      @dblclick="maximizePanel">
+      <slot name="grabber">
+        <va-divider
+          class="fill-width"
+          :vertical="!$props.vertical" />
+      </slot>
+    </div>
+    <div
+      class="va-split__panel"
+      :style="getPanelStyle('end')">
+      <slot name="end" />
+    </div>
+  </section>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType, ref, shallowRef, computed, watch, onMounted } from 'vue'
+import isString from 'lodash/isString.js'
+import isNumber from 'lodash/isNumber.js'
+
+import { useBem, useComponentPresetProp, useStateful, useStatefulEmits, useStatefulProps, useResizeObserver } from '../../composables'
+import { useSplitDragger, useSplitDraggerProps } from './useSplitDragger'
+
+import { warn } from '../../services/utils'
+
+import { VaDivider } from '../va-divider'
+
+type SplitLimit = number | string | string[]
+
+export default defineComponent({
+  name: 'VaSplit',
+
+  components: { VaDivider },
+
+  props: {
+    ...useComponentPresetProp,
+    ...useSplitDraggerProps,
+    ...useStatefulProps,
+    modelValue: {
+      type: Number,
+      default: 50,
+      validator: (v: number) => v <= 100,
+    },
+    maximization: { type: Boolean, default: false },
+    maximizeStart: { type: Boolean, default: false },
+    limits: {
+      type: Array as any as PropType<[SplitLimit, SplitLimit]>,
+      default: () => ['30%', '30%'],
+    },
+  },
+
+  emits: useStatefulEmits,
+
+  setup: (props, { emit }) => {
+    const splitPanelsContainer = shallowRef<HTMLElement>()
+
+    const { valueComputed } = useStateful(props, emit)
+
+    const containerSize = ref()
+    const getContainerSize = () => {
+      containerSize.value = props.vertical ? splitPanelsContainer.value?.offsetHeight : splitPanelsContainer.value?.offsetWidth
+    }
+    onMounted(getContainerSize)
+    useResizeObserver([splitPanelsContainer], getContainerSize)
+
+    const convertToPercents = (v: string | number) => {
+      let numberValue = ''
+      let measureValue = ''
+
+      if (isNumber(v)) { return v }
+
+      v.split('')
+        .filter((char) => char)
+        .forEach((char) => {
+          !isNaN(+char) ? numberValue += char : measureValue += char
+        })
+
+      switch (measureValue) {
+        case '%':
+          return +numberValue
+        case 'px':
+          return (+numberValue / containerSize.value!) * 100
+        case 'rem':
+          return ((+numberValue * 16) / containerSize.value!) * 100
```

Comment:
it might be wrong if rem is not equal to 16 px

Reply:
Yes, but there is no another source of truth.

Reply:
Mb we can move our base font-size to the var and use it?...

Reply:
SSR problems. Let's leave it for grooming. Because it seems better to delete rem at all.

Reply:
Done.